hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/CloudBigtableReplicator.java

```diff
@@ -273,9 +342,9 @@ private int getRowSize(Map.Entry<ByteRange, List<Cell>> rowCells) {
       String tableName, Map<ByteRange, List<Cell>> batchToReplicate) {
     try {
       CloudBigtableReplicationTask replicationTask =
-          new CloudBigtableReplicationTask(tableName, connection, batchToReplicate);
-      return executorService.submit(replicationTask);
-    } catch (IOException ex) {
+          new CloudBigtableReplicationTask(tableName, sharedResources.connection, batchToReplicate);
+      return sharedResources.executorService.submit(replicationTask);
+    } catch (Exception ex) {
       LOG.error("Failed to submit a batch for table: " + tableName, ex);
```

Comment:
I think you should handle interrupted exception:
if (ex instanceof InterruptedException) {
  Thread.current().interrupt();
}

Reply:
done.